crates/cxx-qt-gen/src/generator/rust/signals.rs

```diff
@@ -11,11 +11,137 @@ use crate::{
     },
-    parser::signals::ParsedSignal,
+    parser::{mappings::ParsedCxxMappings, signals::ParsedSignal},
 };
-use quote::quote;
+use quote::{format_ident, quote};
 use syn::{parse_quote, FnArg, Ident, Path, Result};
 
+pub fn generate_rust_free_signal(
+    signal: &ParsedSignal,
+    cxx_mappings: &ParsedCxxMappings,
+    module_ident: &Ident,
+) -> Result<GeneratedRustQObject> {
+    let qobject_name = &signal.qobject_ident;
+    let idents = QSignalName::from(signal);
+    let signal_name_rust = idents.name.rust;
+    let signal_name_rust_str = signal_name_rust.to_string();
+    let signal_name_cpp = idents.name.cpp;
+    let signal_name_cpp_str = signal_name_cpp.to_string();
+    let free_connect_ident_cpp =
+        format_ident!("{}_{}", signal.qobject_ident, idents.connect_name.cpp);
+    let free_connect_ident_rust =
+        format_ident!("{}_{}", signal.qobject_ident, idents.connect_name.rust);
+    let free_connect_ident_rust_str =
+        format_ident!("{}_{}", signal.qobject_ident, idents.connect_name.rust);
+    let connect_ident_rust = idents.connect_name.rust;
+    let on_ident_rust = idents.on_name;
+
+    // Build a namespace that includes any namespace for the T
+    let connect_namespace = QSignalName::free_namespace_from_qobject_namespace(
+        cxx_mappings
+            .namespaces
+            .get(&signal.qobject_ident.to_string()),
+    );
+
+    let parameters_cxx: Vec<FnArg> = signal
+        .parameters
+        .iter()
+        .map(|parameter| {
+            let ident = &parameter.ident;
+            let ty = &parameter.ty;
+            parse_quote! { #ident: #ty }
+        })
+        .collect();
+    let parameters_qualified: Vec<FnArg> = parameters_cxx
+        .iter()
+        .cloned()
+        .map(|mut parameter| {
+            if let FnArg::Typed(pat_type) = &mut parameter {
+                *pat_type.ty =
+                    syn_type_cxx_bridge_to_qualified(&pat_type.ty, &cxx_mappings.qualified);
+            }
+            parameter
+        })
+        .collect();
+
+    let self_type_cxx = if signal.mutable {
+        parse_quote! { Pin<&mut #qobject_name> }
+    } else {
+        parse_quote! { &#qobject_name }
+    };
+    let self_type_qualified =
+        syn_type_cxx_bridge_to_qualified(&self_type_cxx, &cxx_mappings.qualified);
+    let qualified_impl =
+        syn_ident_cxx_bridge_to_qualified_impl(qobject_name, &cxx_mappings.qualified);
+
+    let mut unsafe_block = None;
+    let mut unsafe_call = Some(quote! { unsafe });
+    if signal.safe {
+        std::mem::swap(&mut unsafe_call, &mut unsafe_block);
+    }
+
+    let attrs = &signal.method.attrs;
+
+    let fragment = RustFragmentPair {
+        cxx_bridge: vec![
+            quote! {
+                #unsafe_block extern "C++" {
+                    #(#attrs)*
+                    #[rust_name = #signal_name_rust_str]
+                    #unsafe_call fn #signal_name_cpp(self: #self_type_cxx, #(#parameters_cxx),*);
```

Comment:
Doesn't this basically re-generate the declaration the user wrote initially? As emitting the signal is always just a normal function call that we need to bind to.

Could you try just removing the `#[qsignal]` specifier and then adding the ForeignItemFn the the passthrough_items?
As then we could get rid of this generation here.

Reply:
I think we can in this case potentially easily ... Note that #616 is where i'll look further into it for other places as well.

I'll see if i can make the parser happy for the generation here.

Reply:
done